clash-lib/src/Clash/Netlist/Util.hs

```diff
@@ -379,6 +379,7 @@ typeSize :: HWType
          -> Int
 typeSize (Void {}) = 0
 typeSize String = 1
+typeSize Integer = 1
```

Comment:
I'm confused. Shouldn't this be whatever integer width Clash is provided through `-fclash-intwidth`? (Also see other comment)

Reply:
Given that's internal use only, and doesn't actually a represent a Haskell `Int` or `Integer`, `0` is probably the better value. For historical reasons, the `typeSize` of `String` is `1`, but it should probably be made `0` as well.

This should be added as a comment.